lib/cassandra/cluster/control_connection.rb

```diff
@@ -825,7 +826,7 @@ def refresh_schema_async_wrapper
         # 2. avoid race conditions where those updates may return older data than our
         #    full refresh and might win as last writer with that potentially older data.
         synchronize do
-          if @refresh_schema_future
+          if defined?(@refresh_schema_future) && @refresh_schema_future
```

Comment:
@avalanche123 happy to. `@refresh_schema_future` seemed much more private than the rest of the variables in the initializer. You prefer that?